src/app/features/worklog/worklog-export/worklog-export.component.ts

```diff
@@ -350,6 +361,13 @@ export class WorklogExportComponent implements OnInit, OnDestroy {
         group.notes[i] = group.notes[i].replace(/\n/g, ' - ');
       }
 
+      group.projects = unique(group.projects).filter((projectId: string) => !!projectId);
+      for (let i = 0; i < group.projects.length; i++) {
+        this._projectService.getByIdOnce$(group.projects[i]).subscribe(project => {
```

Comment:
As mentioned in the mail: We're mixing async and synchronous code here, which will not work well. An option would be to get all the projects and all the tags in the observable on line 152 and to map them here.

Reply:
that is exactly what I suspected. I just pushed another commit with your suggested changes. thank you